browser/src/control/Parts.js

```diff
@@ -478,21 +478,20 @@ L.Map.include({
 	},
 
 	showPage: function () {
-		if (this.getDocType() === 'spreadsheet' && this.hasAnyHiddenPart()) {
-			var partNames_ = this._docLayer._partNames;
-			var hiddenParts_ = this._docLayer._hiddenParts;
+		if (this.getDocType() === 'spreadsheet' && app.calc.isAnyPartHidden()) {
+			var hiddenParts = app.calc.getHiddenPartNameArray();
 
-			if (hiddenParts_.length > 0) {
+			if (app.calc.isAnyPartHidden()) {
 				var container = document.createElement('div');
 				container.style.maxHeight = '300px';
 				container.style.overflowY = 'auto';
-				for (var i = 0; i < hiddenParts_.length; i++) {
+				for (var i = 0; i < hiddenParts.length; i++) {
 					var checkbox = document.createElement('input');
 					checkbox.type = 'checkbox';
-					checkbox.id = 'hidden-part-checkbox-' + String(hiddenParts_[i]);
+					checkbox.id = 'hidden-part-checkbox-' + hiddenParts[i];
```

Comment:
what if it has space?

Reply:
Checked it now, it works event it has spaces at the start and end of the sheet name.